test/src/test/java/org/corfudb/integration/CorfuReplicationClusterConfigIT.java

```diff
@@ -0,0 +1,619 @@
+            mapActive.put(String.valueOf(i), i);
+            activeRuntime.getObjectsView().TXEnd();
+        }
+        assertThat(mapActive.size()).isEqualTo(secondBatch);
+
+        // Wait until data is fully replicated again
+        waitForReplication(size -> size == secondBatch, mapStandby, secondBatch);
+        log.info("After delta sync, both maps have size {}. Current active corfu[{}] log tail " +
+                        "is {}, standby corfu[{}] log tail is {}", secondBatch, activeClusterCorfuPort,
+                activeRuntime.getAddressSpaceView().getLogTail(), standbyClusterCorfuPort,
+                standbyRuntime.getAddressSpaceView().getLogTail());
+
+        // Verify data
+        for (int i = 0; i < secondBatch; i++) {
+            assertThat(mapStandby.containsKey(String.valueOf(i))).isTrue();
+        }
+        log.info("Log replication succeeds without config change!");
+
+        // Perform a config update with all standby
+        corfuStore.tx(DefaultClusterManager.CONFIG_NAMESPACE)
+                .update(DefaultClusterManager.CONFIG_TABLE_NAME, DefaultClusterManager.OP_ALL_STANDBY,
+                        DefaultClusterManager.OP_ALL_STANDBY, DefaultClusterManager.OP_ALL_STANDBY)
+                .commit();
+        assertThat(configTable.count()).isOne();
+        log.info("New topology config applied!");
+        TimeUnit.SECONDS.sleep(mediumInterval);
+
+        for (int i = secondBatch; i < thirdBatch; i++) {
+            activeRuntime.getObjectsView().TXBegin();
+            mapActive.put(String.valueOf(i), i);
+            activeRuntime.getObjectsView().TXEnd();
+        }
+        assertThat(mapActive.size()).isEqualTo(thirdBatch);
+        log.info("Active map have {} entries now!", thirdBatch);
+
+        // Standby map should still have secondBatch size
+        log.info("Standby map should still have {} size", secondBatch);
+        waitForReplication(size -> size == secondBatch, mapStandby, secondBatch);
+
+        // Double check after 10 seconds
+        TimeUnit.SECONDS.sleep(mediumInterval);
+        assertThat(mapActive.size()).isEqualTo(thirdBatch);
+        assertThat(mapStandby.size()).isEqualTo(secondBatch);
+    }
+
+    /**
+     * This test verify config change with one active and one invalid
+     * <p>
+     * 1. Init with corfu 9000 active and 9001 standby
+     * 2. Write 10 entries to active map
+     * 3. Start log replication: Node 9010 - active, Node 9020 - standby
+     * 4. Wait for Snapshot Sync, both maps have size 10
+     * 5. Write 5 more entries to active map, to verify Log Entry Sync
+     * 6. Perform a active-invalid config update with corfu store
+     * 7. Write 5 more entries to active map
+     * 8. Verify data will not be replicated, since standby is invalid
+     * 9. Resume to standby and verify data is fully replicated again.
+     */
+    @Test
+    public void testNewConfigWithInvalidClusters() throws Exception {
```

Comment:
If you take a look at PR #2640 I broke all these functions of writing to active/standby maps/ stopping into generic utilities available in LogReplicationAbstractIT, which would reduce a lot of the code here (just a suggestion)

Reply:
A similar suggestion is to move the common code from all tests to a common place in this file so the tests themselves are smaller